### PR TITLE
Add test for checking serial device function.

### PR DESCRIPTION
**Pull request by Hao-Liu**
This test if used for checking basic serial character device function
for various type of serial devices.

1) Define the VM with specified serial device and define result meets expectation.
2) Test whether defined XML meets expectation
3) Start the guest and check if start result meets expectation
4) Test the function of started serial device
5) Shutdown the VM and check whether cleaned up properly
6) Clean up environment

This PR depends on https://github.com/autotest/virt-test/pull/1978 and https://github.com/autotest/virt-test/pull/1979
Signed-off-by: Hao Liu hliu@redhat.com

### CODE REVIEW

**Comment by will-Do** [comment]
There's a module for graphics, so the type of this case need rename
```
$ ls libvirt/tests/src/graphics/
functional.py
```
And check the case log:
```
18:44:11 DEBUG| Searching for test modules that match 'type = functional' and 'provider = ydu-libvirt' on this cartesian dict
18:44:11 DEBUG| Found subtest module /home/ydu/Work/tp-libvirt/libvirt/tests/src/graphics/functional.py
```

BTW, there's some misuse about ```logging.error``` and ```logging.debug```

File: libvirt/tests/src/serial/serial_functional.py [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+            os.close(self.read_fd)
+        if self.write_fd is not None:
+            os.close(self.write_fd)
+        if self.console_type == 'pipe':
+            if os.path.exists(self.address + '.in'):
+                os.remove(self.address + '.in')
+            if os.path.exists(self.address + '.out'):
+                os.remove(self.address + '.out')
+
+
```
Comment (rbian) [inline libvirt/tests/src/serial/serial_functional.py]: What about moving this class to avocado-vt?

Comment (waynesun09) [inline libvirt/tests/src/serial/serial_functional.py]: @Hao-Liu plan to move this to avocado-vt?

**Comment by waynesun09** [comment]
This have been updated and tracking by PR:
https://github.com/autotest/tp-libvirt/pull/1171

close this one now, thanks

**Pull request closed without merging**